docs/interface.rst

```diff
@@ -0,0 +1,106 @@
+Interface
+=========
+
+HTTP API
+--------
+
+Magic-Folder exposes all of its supported functionality in an HTTP API.
+The address of the HTTP server is part of the `daemon configuration`_.
+
+Authorization
+~~~~~~~~~~~~~
+
+The HTTP API is protected by Bearer token-style authorization scheme.
+Only requests which include the correct token will receive successful responses.
+Other requests will receive **401 Unauthorized** responses which omit the requested resource.
+The token value should be included with the **Bearer** scheme in the **Authorization** header element.
+For example::
+
+  Authorization: Bearer abcdefghijklmnopqrstuvwxyz
+
+The correct token value can be found in the *api_token* file inside the Magic-Folder daemon configuration directory.
+The token value is periodically rotated so clients must be prepared to receive an **Unauthorized** response even when supplying the token.
+In this case,
+the client should re-read the token from the filesystem to determine if the value held in memory has become stale.
+
+.. _`daemon configuration`: :ref:`config`
+
+``GET /v1/magic-folder``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This endpoint returns a list of all individual magic-folders managed by this daemon.
+
+The response code **OK** and the **Content-Type** is ``application/json``.
+The response body follows the form of this example::
+
+  { "folders":
+    [ { "name": "Alice's music", "local-path": "/home/alice/Music" }
+    , { "name": "Secret docs", "local-path": /home/alice/secrets" }
```

Comment:
Perhaps not actionable for this PR, but is there a reason we show only these two bits of config? What about the directory-capabilities? Author name, keys?

Internally we call the path `magic_directory` so it might be better to be consistent (versus the state-location versus the stash-path for example).

Reply:
No particular reason, just the two obvious pieces of state.  This should be easy to expand when it becomes clear what's needed - by GridSync, I suppose?  Will GridSync need to know the caps?  I guess it does create some files on the grid somewhere but I don't think it creates them inside the collective or dmd directories ... I can't remember exactly right now.  And maybe it will change anyway?  We should get @crwood involved here.

It does seem like any changes here can wait for a future PR though.  At that point it'll just be small twiddles to the resource, I think.

Reply:
Sure, I guess it can be future. I think it'll need to export ALL the information (e.g. to support moving the CLI over, which does list all those things). For example, via `magic-folder list --include-secret-information`